src/java.base/share/classes/java/util/Locale.java

```diff
@@ -1069,30 +1074,29 @@ private static Locale initDefault() {
             }
             script = "";
         } else {
-            script = props.getProperty("user.script", "");
-            country = props.getProperty("user.country", "");
-            variant = props.getProperty("user.variant", "");
+            script = StaticProperty.userScript(0);
+            country = StaticProperty.userCountry(0);
+            variant = StaticProperty.userVariant(0);
         }
 
         return getInstance(language, script, country, variant,
-                getDefaultExtensions(props.getProperty("user.extensions", ""))
+                getDefaultExtensions(StaticProperty.userExtensions(0))
                     .orElse(null));
     }
 
+    @SuppressWarnings("removal")
     private static Locale initDefault(Locale.Category category) {
-        Properties props = GetPropertyAction.privilegedGetProperties();
-
         Locale locale = Locale.defaultLocale;
+        var sm = System.getSecurityManager();
+        if (sm != null) {
+            sm.checkPropertiesAccess();
+        }
         return getInstance(
-            props.getProperty(category.languageKey,
-                    locale.getLanguage()),
-            props.getProperty(category.scriptKey,
-                    locale.getScript()),
-            props.getProperty(category.countryKey,
-                    locale.getCountry()),
-            props.getProperty(category.variantKey,
-                    locale.getVariant()),
-            getDefaultExtensions(props.getProperty(category.extensionsKey, ""))
+            StaticProperty.userLanguage(category.ordinal() + 1),
+            StaticProperty.userScript(category.ordinal() + 1),
+            StaticProperty.userCountry(category.ordinal() + 1),
+            StaticProperty.userVariant(category.ordinal() + 1),
+            getDefaultExtensions(StaticProperty.userExtensions(category.ordinal() + 1))
```

Comment:
I suspect this would be more readable, and maintainable, if you add non-public languageKey/scriptKey/countryKey/variantKey methods to Locale.Category. That would change this to getInstance(category.languageKey(), ...) so no sight of the ordinal at the use-site.